packages/amplify-category-hosting/lib/S3AndCloudFront/helpers/file-uploader.js

```diff
@@ -19,7 +19,13 @@ async function run(context, distributionDirPath) {
 
   const hasCloudFront = !!context?.exeInfo?.template?.Resources?.CloudFrontDistribution;
 
-  fileList.forEach(filePath => {
+
+  await Promise.all(fileList.map(async filePath => {
+    if(!filePath.includes('index.html')) {
+      return await uploadFileTasks.push(() => uploadFile(s3Client, hostingBucketName, distributionDirPath, filePath, hasCloudFront));
```

Comment:
nit: I don't think this `await` is needed since the Promise.all is being awaited

```suggestion
      return uploadFileTasks.push(() => uploadFile(s3Client, hostingBucketName, distributionDirPath, filePath, hasCloudFront));
```

Reply:
Done 3abfb5164c510f1697bc828246313a66f817646e